internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -524,20 +531,49 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
 
 	entryPoint := w.data.Session.EntryPoint()
 
-	run, err := sdkruntimes.Run(
-		ctx,
-		sdkruntimes.RunParams{
-			Runtimes:             w.ws.svcs.Runtimes,
-			BuildFile:            w.data.BuildFile,
-			Globals:              w.globals,
-			RunID:                runID,
-			FallthroughCallbacks: cbs,
-			EntryPointPath:       entryPoint.Path(),
-		},
-	)
-	if err != nil {
-		// TODO(ENG-130): discriminate between infra and real errors.
-		return prints, err
+	type runOrErr struct {
+		run sdkservices.Run
+		err error
+	}
+
+	runDone := make(chan runOrErr)
```

Comment:
Should make this buffered to avoid goroutine leak

Reply:
https://github.com/autokitteh/autokitteh/pull/629